edgedb/asyncio_con.py

```diff
@@ -32,41 +35,160 @@
 from .datatypes import datatypes
 from .protocol import asyncio_proto
 from .protocol import protocol
+from .protocol.protocol import CodecsRegistry as _CodecsRegistry
+from .protocol.protocol import QueryCodecsCache as _QueryCodecsCache
 
 
-class _ConnectionProxy:
-    # Base class to enable `isinstance(AsyncIOConnection)` check.
-    __slots__ = ()
+ERRNO_RE = re.compile(r"\[Errno (\d+)\]")
+TEMPORARY_ERRORS = frozenset({
+    errno.ECONNREFUSED,
+    errno.ECONNABORTED,
+    errno.ECONNRESET,
+    errno.ENOENT,
+})
 
 
-class AsyncIOConnectionMeta(abc.ABCMeta):
+def extract_errno(s):
+    result = []
+    for match in ERRNO_RE.finditer(s):
+        result.append(int(match.group(1)))
+    if result:
+        return result
+
+
+class _AsyncIOConnectionImpl:
+
+    def __init__(self, codecs_registry, query_cache):
+        self._codecs_registry = codecs_registry
+        self._query_cache = query_cache
+
+        # TODO(tailhook) temporary, this is not needed when we implement
+        # transaction part of RFC1004
+        self._top_xact = None
 
-    def __instancecheck__(cls, instance):
```

Comment:
Why was this deleted?

Reply:
Because it's now implemented via `abstract.AsyncIOExecutor`, right?

Reply:
No. Because `PoolConnection` is a subclass of `AsyncIOConnection` anyway. And we use two hooks (most notably `_reconnect`) to make it working for pools instead of using Proxy pattern and metaclasses.